cloudbuild/python-scripts/parse_logs.py

```diff
@@ -324,34 +354,44 @@ def read_logs(cluster_temp_bucket, logs_pattern, query):
     # Check if metric is present in at least one of the files.
     is_metric_found = False
     is_query_result_found = False
-    # Get all the contents in the GCS Bucket.
-    gcs_bucket_contents = get_bucket_contents(cluster_temp_bucket)
+    is_unbounded_result_found = False
 
     # Form the logs_pattern.
     # logs are stored in files having names of the format 'log_pattern/...'
     # the pattern enables searching for the same.
     logs_pattern = re.compile(rf'{re.escape(logs_pattern)}/.+')
 
+    # Get all the contents in the GCS Bucket.
+    gcs_bucket_contents = get_bucket_contents(cluster_temp_bucket)
     # Find all strings in the array that match the pattern.
     gcs_log_objects = [
         gcs_log_object
         for gcs_log_object in gcs_bucket_contents
         if logs_pattern.match(gcs_log_object)
     ]
+
     for gcs_log_object in gcs_log_objects:
+        logging.info(f'Found logs file: {gcs_log_object}')
         # -1 is returned in case metric not found in the log file.
-        (metric_value, is_query_result_present) = get_blob_and_check_metric(
+        (metric_value, is_query_result_present,
+         is_unbounded_result_present) = get_blob_and_check_metric(
             gcs_log_object,
             cluster_temp_bucket,
             metric_string,
             end_of_metric_string,
-            query,
+            query, mode
         )
         if metric_value != -1:
             is_metric_found = True
             # Sum up all the values.
             total_metric_count += metric_value
 
+        # Check if unbounded read result is found in any one of the files.
+        # True if found in at least one of the files.
+        # False in case not found in any.
+        if is_unbounded_result_present:
+            is_unbounded_result_found = True
```

Comment:
These flags seem redundant. Just `is_unbounded_result_present` can be used without introducing `is_unbounded_result_found `.

Reply:
`is_unbounded_result_present` is the local variable that gets the result from one log file. `is_unbounded_result_found ` is an OR across all the log files as we need the result to be present in atleast one log file.

Reply:
Got it